coordinator/stream.go

```diff
@@ -426,29 +417,26 @@ func (s *Stream) GenerateGroupKey(ctx *streamCtx, keys []string, value *influx.R
 	if len(keys) == 0 {
 		return ""
 	}
-	builder := ctx.bp.Get()
-	defer func() {
-		builder.Reset()
-		ctx.bp.Put(builder)
-	}()
-
-	tagIndex := 0
-	for i := range keys {
-		idx := util.Search(tagIndex, len(value.Tags), func(j int) bool { return value.Tags[j].Key >= keys[i] })
-		if idx < len(value.Tags) && value.Tags[idx].Key == keys[i] {
-			builder.AppendString(value.Tags[idx].Value)
-			if i < len(keys)-1 {
-				builder.AppendByte(config.StreamGroupValueSeparator)
-			}
-			tagIndex = idx + 1
-			continue
+
+	values := make([]string, len(keys))
```

Comment:
The builder involves memory reuse. It is recommended that benchmarks be used to quantify the performance before and after the comparison.

Reply:
I think memory reuse is better. When the write traffic is relatively large, continuously applying and releasing memory has a certain impact on the performance. What is the consideration for restructuring this part of the code?

Reply:
I'm changing these lines because I removed the `StringBuilder`, its implemnation is not robust enough and may cause issues in long term maintenance.

I've noticed the memory reuse, to minimize the performance punishment, I refactored the code to use `strings.Join` instead of `append`.

but I haven't done a benchmark to find out the performance difference, I will try to get a result when I have time.

Reply:
the benchmark shows there's some performance decrease, I will try to find another way to optimize.
and I will close this PR and open seperate PRs because the code changes is far more than my expectation and still increasing when I try to resolve the review comments.